entrypoints/setup_git.sh

```diff
@@ -1,24 +1,20 @@
 #!/bin/sh
 
-apk update && apk add git
-
-git remote || INIT=true
+ls .git && return 0
 
 echo "${GITHUB_REPO}" | grep -q "#" && TAG="${GITHUB_REPO#*#}" || TAG=
 
-if [ "${INIT}" ]
+apk update && apk add git
+git init
+chown -R "$(find . -maxdepth 1 -exec ls -ld {} + | awk '{print $3":"$4}' | tail -n1)" .git
+git config --global --add safe.directory "${PWD}"
+REPO="${GITHUB_REPO%%#*}"
+git remote add origin "${REPO}"
+git fetch
+DEFAULT_BRANCH=$(git remote show origin | sed -n '/HEAD branch/s/.*: //p')
+git reset --hard origin/"${DEFAULT_BRANCH}"
```

Comment:
this will inherit the changes of the other PR, so it will only be run once at container creation (also it only runs if git is installed for the first time, as the `ls .git && return 0` would make it exit if already there). But in general the `reset` is needed because `git init` only adds the `.git` folder but does not track files (so they would all appear new). This is to sync with the remote status.

Reply:
also, maybe worth noticing that this PR only replaces the IF with the return 0. The diff shows quite some changes but they are mostly spaces (`hide whitespaces` in github makes it a lot clearer)

Reply:
ok I get it now. It is more a test area for the other PRs, not an environment where we want to push back . Happy to approve